### ISSUE TITLE
no-return-await doesn't respect boolean operators.

### ISSUE THREAD

**Issue by scriptcoded** [eslint/eslint#10835]
**Tell us about your environment**

* **ESLint Version:** 5.5.0
* **Node Version:** 10.4.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
StandardJS

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async function test (a, b) {
  return await a || await b
}
```

This was run over at https://eslint.org/demo/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gdGVzdCAoYSwgYikge1xyXG5cdHJldHVybiBhIHx8IGF3YWl0IGJcclxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6eyJleHBlcmltZW50YWxPYmplY3RSZXN0U3ByZWFkIjp0cnVlfX0sInJ1bGVzIjp7Im5vLXJldHVybi1hd2FpdCI6Mn0sImVudiI6e319fQ==

**What did you expect to happen?**
No errors should appear.

**What actually happened? Please include the actual, raw output from ESLint.**
no-return-await says that `await` should be removed from the return statement. However, that would return the first promise. It is the results of `a` and `b` that should be compared.

**Comment by platinumazure** [comment]
Hi @scriptcoded, thanks for the issue.

I think what the rule wants you to do here is to get the await expression out of the return statement:

```js
async function test (a, b) {
  const value = await a || await b
  return value
}
```

So in that sense, the rule is working as designed.

However, if you think the error message can be improved, we would welcome any suggestions (or a PR) to that effect.

Let me know if you think I'm misunderstanding the issue.

**Comment by not-an-aardvark** [comment]
This could be replaced with:

```js
async function test (a, b) {
  return await a || b
}
```

...and the behavior of the code would be the same.

**Comment by scriptcoded** [comment]
@not-an-aardvark No, that would await only `a`

**Comment by scriptcoded** [comment]
@platinumazure That would indeed be true, but that shouldn't be necessary. Take this example:

```javascript
async p => (
  await p.a() ||
  await p.b()
)
```

In this case I need the syntax to be short and concise. Since the arrow function has an explicit return value, I can't split it into multiple lines.

**Comment by not-an-aardvark** [comment]
> No, that would await only `a`

The code would have the same behavior because when a Promise like `b` is returned, it's flattened as if `await` is implicitly called on it.

**Comment by scriptcoded** [comment]
@not-an-aardvark Yes, I see now. My bad. Thank you for your patience!
